Geo/11003-ManageMultipleMovingPointLoads

mcgicjn2 opened the pull request:
**📝 Description**
Create process managing a collection of moving_loads

- initialization, operations must be consistent
- multiple moving loads based on a configuration translating to an offset

This will be a PR depends on the structuralmechanics application until PR #10997 is merged

Closes #11003 

**🆕 Changelog**
Added process to set and manage a collection multiple moving point loads
- clones mutliple moving_line_conditions into compute model part
- removes associated original moving line conditions from compute model part
- manages list of multiple set_moving_line_load processes + calls
+ some associated tests

mcgicjn2 committed:
Added process to generate multiple moving loads

Added process to set and manage a collection multiple moving point loads
- clones mutliple moving_line_conditions into compute model part
- removes associated original moving line conditions from compute model part
- manages list of multiple set_moving_line_load processes + calls
+ some associated tests

mcgicjn2 committed:
Update test_set_multiple_moving_load_process.py

Removed trailing white space from python tests

mcgicjn2 committed:
Update set_multiple_moving_loads.py

Corrected set_multiple_moving_loads python file.

mcgicjn2 committed:
Update set_multiple_moving_loads.py

Corrected typo

mcgicjn2 committed:
Update set_multiple_moving_loads.h

Removed redundant SetMultipleMovingLoadsProcess:: in header file

mcgicjn2 committed:
Merge branch 'master' into geo/11003-ManageMultipleMovingPointLoads

mcgicjn2 committed:
Removed trailing white spaces

huhaas commented on applications/GeoMechanicsApplication/custom_processes/set_multiple_moving_loads.cpp:
```diff
@@ -0,0 +1,152 @@
+
+            auto parameters_moving_load = mParameters.Clone();
+
+    		count++;
```
Indents are not consistent due to tab usage. In multiple files.

huhaas commented on applications/GeoMechanicsApplication/custom_processes/set_multiple_moving_loads.h:
```diff
@@ -0,0 +1,111 @@
+
+	        ModelPart& mrModelPart;
+	        Parameters mParameters;
+	        std::vector<SetMovingLoadProcess> mMovingPointLoadsProcesses;
```
Maybe it is better to use the Kratos unique pointer construction here. Currently the push_back in cpp line 80 creates unwanted copies.

huhaas commented on applications/GeoMechanicsApplication/custom_processes/set_multiple_moving_loads.cpp:
```diff
@@ -0,0 +1,152 @@
+
+    }
+
+    ModelPart& SetMultipleMovingLoadsProcess::CloneMovingConditionInComputeModelPart(std::string NewBodyPartName)
```
Argument can be const&

huhaas commented on applications/GeoMechanicsApplication/tests/test_set_multiple_moving_load_process.py:
```diff
@@ -0,0 +1,1886 @@
+import KratosMultiphysics.GeoMechanicsApplication as GMA
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+
+class TestSetMultipleMovingLoadsProcess(KratosUnittest.TestCase):
```
I have some suggestions to improve this class. Let's discuss this in the office. Also SonarCloud does not agree with this change.

huhaas reviewed (changes requested):
Please update

mcgicjn2 committed:
reduced code smells / repeatition

mcgicjn2 committed:
reduced code smells / repetition - part 2

mcgicjn2 committed:
Further improvements towards code bugs and repetition

huhaas commented on applications/GeoMechanicsApplication/tests/test_set_multiple_moving_load_process.py:
```diff
@@ -26,18 +26,20 @@ def setUp(self):
+            tols = [None] * len(rhs)
+        
+        for rhs_val, expected_val, tols in zip(rhs, expected_res, tols):
+            self.assertAlmostEqual(rhs_val, expected_val, tols)
```
```suggestion
        for rhs_val, expected_val, tol in zip(rhs, expected_res, tols):
            self.assertAlmostEqual(rhs_val, expected_val, tol)
```

avdg81 committed:
Fixed two code smells reported by SonarCloud

Extracted two repeated strings and gave them a name.

avdg81 committed:
Merge remote-tracking branch 'origin/master' into geo/11003-ManageMultipleMovingPointLoads

mcgicjn2 committed:
Added review suggestions from HuHaas

mcgicjn2 committed:
Merge branch 'geo/11003-ManageMultipleMovingPointLoads' of https://github.com/KratosMultiphysics/Kratos into geo/11003-ManageMultipleMovingPointLoads

avdg81 committed:
Fixed Codacy issue: removed trailing whitespace

mcgicjn2 committed:
Removed Trailing edges

mcgicjn2 committed:
Merge branch 'geo/11003-ManageMultipleMovingPointLoads' of https://github.com/KratosMultiphysics/Kratos into geo/11003-ManageMultipleMovingPointLoads

mcgicjn2 committed:
Renamed typo in loop variable: tols -> tol

mcgicjn2 committed:
Added vector of process pointers, instead of a pointer to a vector or processes

mcgicjn2 committed:
Merge branch 'master' into geo/11003-ManageMultipleMovingPointLoads

mcgicjn2 committed:
merged master into branch

Commit:
Protect SetMultipleMovingLoadsProcess against unwanted copying

avdg81 committed:
Review feedback by Anne

- Changed three references to references-to-const.
- Made a getter `const`.
- Removed commented out code.
- Removed two empty statements.
- Removed redundant test wrapper methods.
- Fixed two typos.
- Several minor stylistic changes.

avdg81 reviewed (approved):
I have made several minor coding improvements (see my last commit). It looks good to me now, Jon. Thanks for the hard work that you've put into this!
